code/__HELPERS/path.dm

```diff
@@ -0,0 +1,379 @@
+
+		switch(heading)
+			if(NORTHWEST)
+				if(STEP_NOT_HERE_BUT_THERE(current_turf, EAST, NORTHEAST) || STEP_NOT_HERE_BUT_THERE(current_turf, SOUTH, SOUTHWEST))
+					interesting = TRUE
+				else
+					lateral_scan_spec(current_turf, WEST)
+					lateral_scan_spec(current_turf, NORTH)
+			if(NORTHEAST)
+				if(STEP_NOT_HERE_BUT_THERE(current_turf, WEST, NORTHWEST) || STEP_NOT_HERE_BUT_THERE(current_turf, SOUTH, SOUTHEAST))
+					interesting = TRUE
+				else
+					lateral_scan_spec(current_turf, EAST)
+					lateral_scan_spec(current_turf, NORTH)
+			if(SOUTHWEST)
+				if(STEP_NOT_HERE_BUT_THERE(current_turf, WEST, SOUTHEAST) || STEP_NOT_HERE_BUT_THERE(current_turf, NORTH, NORTHWEST))
+					interesting = TRUE
+					return
+				else
+					lateral_scan_spec(current_turf, SOUTH)
+					lateral_scan_spec(current_turf, WEST)
+			if(SOUTHEAST)
+				if(STEP_NOT_HERE_BUT_THERE(current_turf, WEST, SOUTHWEST) || STEP_NOT_HERE_BUT_THERE(current_turf, NORTH, NORTHEAST))
+					interesting = TRUE
+				else
+					lateral_scan_spec(current_turf, SOUTH)
+					lateral_scan_spec(current_turf, EAST)
+
+		if(interesting)
+			var/datum/jps_node/newnode = new(current_turf, unwind_node, steps_taken)
+			openc[current_turf] = newnode
+			open.Insert(newnode)
+			return
+
+#undef PATH_START
+#undef CAN_STEP
+#undef STEP_NOT_HERE_BUT_THERE
+
+// and then the rest are holdovers from the A* file
+
+// These two defines are used for turf adjacency directional window nonsense
+#define MASK_ODD 85
+#define MASK_EVEN 170
+
+/**
+ * Returns adjacent turfs to this turf that are reachable, in all 8 directions
+ *
+ * Arguments:
+ * * caller: The atom, if one exists, being used for mobility checks to see what tiles it can reach
+ * * ID: An ID card that decides if we can gain access to doors that would otherwise block a turf
+ * * simulated_only: Do we only worry about turfs with simulated atmos, most notably things that aren't space?
+*/
+/turf/proc/reachableAdjacentTurfs(caller, ID, simulated_only)
+	var/list/L = new()
+	var/turf/T
+	var/static/space_type_cache = typecacheof(/turf/open/space)
+
+	for(var/iter_dir in GLOB.alldirs)
+		T = get_step(src,iter_dir)
+		if(!T || (simulated_only && space_type_cache[T.type]))
+			continue
+		if(!T.density && !LinkBlockedWithAccess(T,caller, ID))
+			L.Add(T)
+	return L
+
+//Returns adjacent turfs in cardinal directions that are reachable via atmos
+/turf/proc/reachableAdjacentAtmosTurfs()
+	return atmos_adjacent_turfs
+
+/turf/proc/LinkBlockedWithAccess(turf/T, caller, ID)
+	var/adir = get_dir(src, T)
+	var/rdir = ((adir & MASK_ODD)<<1)|((adir & MASK_EVEN)>>1)
+	for(var/obj/structure/window/W in src)
+		if(!W.CanAStarPass(ID, adir))
+			return TRUE
+	for(var/obj/machinery/door/window/W in src)
+		if(!W.CanAStarPass(ID, adir))
+			return TRUE
+	for(var/obj/O in T)
+		if(!O.CanAStarPass(ID, rdir, caller))
+			return TRUE
+
+	return FALSE
```

Comment:
```suggestion
/turf/proc/link_blocked_with_access(turf/neighbor, caller, ID)
	var/actual_dir = get_dir(src, neighbor)

	for(var/obj/structure/window/window in src)
		if(!window.CanAStarPass(ID, actual_dir))
			return TRUE
	for(var/obj/machinery/door/window/windoor in src)
		if(!windoor.CanAStarPass(ID, actual_dir))
			return TRUE
	
	var/reverse_dir = REVERSE_DIR(actual_dir)
	for(var/obj/object in neighbor)
		if(!object.CanAStarPass(ID, reverse_dir, caller))
			return TRUE

	return FALSE
```

You could move `MASK_ODD` and `MASK_EVEN` to the definition of `REVERSE_DIR` if you wanted to add clarity there, since it's no longer needed here.